client/sites-dashboard/components/link-in-bio-banner/link-in-bio-banner-parts.tsx

```diff
@@ -0,0 +1,77 @@
+//todo use variables where possible and check fontSize use in code base to make sure consistent
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import styled from '@emotion/styled';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { close } from '@wordpress/icons';
+import { ComponentProps } from 'react';
+import { useDispatch } from 'react-redux';
+import { savePreference } from 'calypso/state/preferences/actions';
+
+export const LINK_IN_BIO_BANNER_PREFERENCE = `link-in-bio-banner`;
+
+export const Root = styled( 'div' )( {
+	position: 'relative',
+	display: 'flex',
+	backgroundColor: '#D0CCE3',
+	borderRadius: '3px',
+} );
+
+export const Title = styled( 'h3' )( {
+	fontWeight: 500,
+	fontSize: 20,
+	color: '#000000',
+	margin: 0,
+} );
+
+Title.defaultProps = { children: __( 'Your digital identity' ) };
+
+export const Description = styled( 'p' )( {
+	margin: 0,
+} );
+
+Description.defaultProps = {
+	children: __( 'Show the world what you have to offer with a Link to Bio site.' ),
+};
+
+export const DismissButton = ( props: ComponentProps< typeof Button > ) => {
+	const dispatch = useDispatch();
+
+	const handleDismissBanner = () => {
+		recordTracksEvent( 'calypso_link_in_bio_banner_dismiss_click' );
+		dispatch( savePreference( LINK_IN_BIO_BANNER_PREFERENCE, false ) );
+	};
+	return (
+		<Button
+			icon={ close }
+			onClick={ handleDismissBanner }
+			{ ...props }
+			className={ 'dismiss-button' }
+		/>
+	);
+};
+
+export const Image = ( { src }: { src: string } ) => {
+	return <img src={ src } alt={ __( 'Link to Bio banner image' ) } className={ 'banner-image' } />;
+};
+
+const handleBannerCtaClick = () => {
+	recordTracksEvent( 'calypso_link_in_bio_banner_cta_click' );
+};
+
+export const CreateButton = () => {
+	return (
```

Comment:
We could move the tracking inside the button.

Import `useEffect` and apply:
```suggestion
	useEffect( () => {
		recordTracksEvent( 'calypso_link_in_bio_banner_viewed' );
	}, [] );
	
	return (
```

Finally, remove the function `handleBannerViewed` and all its calls:
- https://github.com/Automattic/wp-calypso/blob/76c7c9497e40fc9a402931ed3cd20812dea3df09/client/sites-dashboard/components/link-in-bio-banner/link-in-bio-banner-parts.tsx#L75-L77
- https://github.com/Automattic/wp-calypso/blob/df99f2886e7f491f0ade4a6ffefa014d3d6b73bf/client/sites-dashboard/components/link-in-bio-banner/link-in-bio-double-tile-banner.tsx#L53
- https://github.com/Automattic/wp-calypso/blob/df99f2886e7f491f0ade4a6ffefa014d3d6b73bf/client/sites-dashboard/components/link-in-bio-banner/link-in-bio-row-banner.tsx#L26
- https://github.com/Automattic/wp-calypso/blob/df99f2886e7f491f0ade4a6ffefa014d3d6b73bf/client/sites-dashboard/components/link-in-bio-banner/link-in-bio-tile-banner.tsx#L33

Another option would be to refactor the function into a new custom hook `useBannerViewed`.

Reply:
Ok, I think that's worthwhile, so will make the change. 

One other thing, if the user toggles between list and and grid mode, the viewed stat will get incremented twice. Are we ok with this?

Reply:
That's an interesting point. Most of our "impression" or "viewed" tracks events have the same issue, they naively record an event on mount. So I'd say it's fine because it's a common issue for all of these sorts of events.

I don't think it generally causes an issue because if you use the Funnels tool (PCYsg-5hd-p2) it will only take notice of the first event anyway.